src/us/kbase/auth2/lib/Authentication.java

```diff
@@ -423,12 +432,15 @@ public void localPasswordChange(
 			final UserName userName,
 			final Password password,
 			final Password pwdnew)
-			throws AuthenticationException, UnauthorizedException, AuthStorageException {
-		//TODO PWD do any cross pwd checks like checking they're not the same
+					throws AuthenticationException, UnauthorizedException, AuthStorageException, IllegalPasswordException {
 		byte[] salt = null;
 		byte[] passwordHash = null;
 		try {
 			nonNull(pwdnew, "pwdnew");
+			if(pwdnew.equals(password)) {
+				throw new IllegalPasswordException("Old and new passwords are identical.");
+			}
+			pwdnew.checkValidity();
 			getLocalUser(userName, password); //checks pwd validity and nulls
 			salt = randGen.generateSalt();
 			passwordHash = pwdcrypt.getEncryptedPassword(pwdnew.getPassword(), salt);
```

Comment:
Need to clear the new array created here. Search for getPassword() and check it's being cleared afterwards.

Should probably also search for the constructor and make sure the arrays are being cleared (unless they come from a String in which case there's no point)